### PR TITLE
refactor: Use io.Writer, really buffered cmd exec and use interface for `go list`

### PR DESCRIPTION
**Pull request by arajkumar**
- Use io.Writer to write Manifest
- Use generic interface to execute `go list`. It expose io.ReadCloser for efficient stdout streaming and Wait for the command execution to complete.
- Use error channel to fetch error from concurrent JSON streaming decoder

### CODE REVIEW

File: gomanifest/internal/deps.go [hunk 1]
```diff
@@ -28,68 +29,54 @@ type DepPackage struct {
...
-// Run ... Actual function that executes go list command and returns output as string.
-func (goListCmd GoListCmd) Run() (io.ReadCloser, error) {
-	GoListGoListDeps := exec.Command("go", "list", "-json", "-deps", "./...")
-	GoListGoListDeps.Dir = goListCmd.CWD
-	defer GoListGoListDeps.Start()
-	GoListGoListDeps.Wait()
-	return GoListGoListDeps.StdoutPipe()
-}
-
 // GoListCmdInterface ... Interface to be implemented to execute go list command.
```
Comment (dgpatelgit) [inline gomanifest/internal/deps.go:32]: Comment to be modified, "GoListCmdInterface" to "GoList"

File: gomanifest/internal/deps.go [hunk 2]
```diff
@@ -28,68 +29,54 @@ type DepPackage struct {
...
-type GoListCmdInterface interface {
-	Run() (io.ReadCloser, error)
-}
-
-// GoList ... Structure that handle go list data and extract required packages.
-type GoList struct {
-	Command GoListCmdInterface
+type GoList interface {
```
Comment (dgpatelgit) [inline gomanifest/internal/deps.go:33]: This interface belong to golist, can be moved to golist.go file.

Comment (arajkumar) [reply gomanifest/internal/deps.go:33]: golist.go is an implementation of GoList interface. Usually generic types should have concrete dependency.

**Comment by dgpatelgit** [comment]
ltgm.

**Review by dgpatelgit — APPROVED** [review]
ltgm

**Pull request merged**